### PR TITLE
Reanimated and GestureHandler upgrade

### PR DESCRIPTION
**Pull request by Inbal-Tish**
## Description
Reanimated and GestureHandler upgrade.
In preparation for the RN73 upgrade we need to upgrade Reanimated (from 3.4.0 to 3.8.1) and GestureHandler (from 2.9.0 to 2.14.1).

The upgrades involved some migrations:
1) Wrapping with GestureHandlerRootView - Modal wrap on iOS too.
2) Wrapping with GestureHandlerRootView when needed - SectionWheelPickerScreen.
3) Remove ts ignore for a bug fix on Reanimated 3.5.0.
4) ts-expect-error where useAnimatedGestureHandler migration is needed (creates bug on PanView on Android with this version).

Known issue - PanView gesture does not work on Android only.

## Changelog
Upgrading Reanimated (3.8.1) and GestureHandler (2.14.1) + upgrade fixes

## Additional info
ticket #4185

### CODE REVIEW

**Review by M-i-k-e-l — COMMENTED** [review]
Marquee - iOS - horizontal not working (need to test on master)
TabController - both - no indicator (line beneath the selected tab item; need to test on master)
Slider - Android - performance is really bad (need to test on master)
ColorPickerScreen - both - need to wrap with `<GestureHandlerRootView style={styles.container}>` (remove the style from ScrollView)
PickerScreen - both - need to wrap with `GestureHandlerRootView`
PanView - **both** - drag does not work

Still did not go over everything

**Comment by Inbal-Tish** [comment]
> Marquee - iOS - horizontal not working (need to test on master) TabController - both - no indicator (line beneath the selected tab item; need to test on master) Slider - Android - performance is really bad (need to test on master) ColorPickerScreen - both - need to wrap with `<GestureHandlerRootView style={styles.container}>` (remove the style from ScrollView) PickerScreen - both - need to wrap with `GestureHandlerRootView` PanView - **both** - drag does not work
> 
> Still did not go over everything

Marquee - doesn't work on master - works on a real device on both branches.
Slider - works fine on both platforms on master and upgrade branches.
TabController - works fine on both platforms on master, on upgrade branch **no indicator**.
PanView - Works fine on both platforms on master, on upgrade branch works fine on iOS, **not on Android**.
PickerScreen, ColorPickerScreen, GridViewScreen - fixed (wrapped with gestureHandlerRootHOC).
GridViewScreen  - looks weird on both masted and upgrade branches.
ProgressiveImageScreen - crashes on both masted and upgrade branches.

**Review by ethanshar — APPROVED** [review]
@M-i-k-e-l 
Wrote small comments, anyway approved

File: src/components/tabController/useScrollToItem.ts [hunk 1]
```diff
@@ -154,9 +147,24 @@ const useScrollToItem = <T extends ScrollToSupportedViews>(props: ScrollToItemPr
...
+      return value;
+    });
+
+    itemsOffsetsAnimated.modify((value) => {
+      'worklet';
+      value.forEach((_, index) => {
+        if (index > 0) {
+          value[index] = value[index - 1] + widths[index - 1];
+        }
+      });
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/tabController/useScrollToItem.ts:162]: It appears that accessing value[index - 1] gives `0`

File: src/components/tabController/useScrollToItem.ts [hunk 2]
```diff
@@ -154,9 +147,24 @@ const useScrollToItem = <T extends ScrollToSupportedViews>(props: ScrollToItemPr
...
-    // trigger value change
-    itemsWidthsAnimated.value = [...itemsWidthsAnimated.value];
-    itemsOffsetsAnimated.value = [...itemsOffsetsAnimated.value];
+    // Update shared values
+    itemsWidthsAnimated.modify((value) => {
+      'worklet';
+      value.forEach((_, index) => {
+        value[index] = widths[index];
+      });
+      return value;
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/tabController/useScrollToItem.ts:156]: Tried it but it caused a TS error I could not find a fix for.

**Pull request merged**